File: app/src/main/java/apps/jizzu/simpletodo/utils/Interpolator.java
package apps.jizzu.simpletodo.utils;

/**
 * Class for getting interpolation for FAB animation.
 */
public class Interpolator implements android.view.animation.Interpolator {

    private double mAmplitude = 1;
    private double mFrequency = 10;

    public Interpolator(double amplitude, double frequency) {
        mAmplitude = amplitude;
        mFrequency = frequency;
    }

    @Override
    public float getInterpolation(float time) {
        return (float) (-1 * Math.pow(Math.E, -time/ mAmplitude) *
                Math.cos(mFrequency * time) + 1);
    }
}

File: app/src/main/java/apps/jizzu/simpletodo/utils/Interpolator.kt
package apps.jizzu.simpletodo.utils

/**
 * Class for getting interpolation for FAB animation.
 */
class Interpolator(private val mAmplitude: Double, private val mFrequency: Double) : android.view.animation.Interpolator {

    override fun getInterpolation(time: Float): Float {
        return (-1.0 * Math.pow(Math.E, -time / mAmplitude) *
                Math.cos(mFrequency * time) + 1).toFloat()
    }
}
